src/rust-crypto/backup.ts

```diff
@@ -166,9 +187,73 @@ export class RustBackupManager extends TypedEventEmitter<RustBackupCryptoEvents,
         this.emit(CryptoEvent.KeyBackupStatus, false);
     }
 
-    private async scheduleAllGroupSessionsForBackup(): Promise<void> {
-        // TODO stub
+    private async backupKeysLoop(maxDelay = 10000): Promise<void> {
+        if (this.backupKeysLoopRunning) {
+            logger.log(`Backup loop already running`);
+            return;
+        }
+        this.backupKeysLoopRunning = true;
+
+        logger.log(`Starting loop for ${this.activeBackupVersion}.`);
+
+        // wait between 0 and `maxDelay` seconds, to avoid backup
+        // requests from different clients hitting the server all at
+        // the same time when a new key is sent
+        const delay = Math.random() * maxDelay;
+        await sleep(delay);
+
+        try {
+            let numFailures = 0; // number of consecutive network failures for exponential backoff
+
+            while (!this.stopped) {
+                // Get a batch of room keys to upload
+                const request: RustSdkCryptoJs.KeysBackupRequest | null = await this.olmMachine.backupRoomKeys();
+
+                if (!request || this.stopped || !this.activeBackupVersion) {
+                    logger.log(`Ending loop for ${this.activeBackupVersion}.`);
+                    return;
+                }
+
+                try {
+                    await this.outgoingRequestProcessor.makeOutgoingRequest(request);
+                    numFailures = 0;
+
+                    const keyCount: RustSdkCryptoJs.RoomKeyCounts = await this.olmMachine.roomKeyCounts();
+                    const remaining = keyCount.total - keyCount.backedUp;
+                    this.emit(CryptoEvent.KeyBackupSessionsRemaining, remaining);
+                } catch (err) {
+                    numFailures++;
+                    logger.error("Error processing backup request for rust crypto-sdk", err);
+                    if ((<MatrixError>err).data) {
```

Comment:
`err` in Typescript is unknown, you should be doing `err instanceof MatrixError` otherwise `null.data` will throw an exception

Reply:
updated